ppa/archive/models.py

```diff
@@ -121,6 +122,76 @@ def stats():
         return stats
 
 
+class ProtectedFlags(Flags):
+    '''Flag set of fields that should be protected if edited in the admin.'''
+    title = ()
+    subtitle = ()
+    sort_title = ()
+    enumcron = ()
+    author = ()
+    pub_place = ()
+    publisher = ()
+    pub_date = ()
+
+    @classmethod
+    def deconstruct(cls):
+        '''Give Django information needed to make
+        :class:`ProtectedField.no_flags` default in migration.'''
+        # (import path, [args], kwargs)
+        return ('ppa.archive.models.ProtectedFlags', ['no_flags'], {})
+
+    @classmethod
+    def all_fields(cls):
+        return list(ProtectedFlags.__members__.keys())
+
+    def as_list(self):
+        '''Return combined flags as a list of field names.'''
+        all_fields = list(self.__members__.keys())
+        return [field for field in all_fields
+                if getattr(self, field)]
+
+    def __str__(self):
```

Comment:
Can't you make use of the `to_simple_str` method on flags objects? Maybe just change `|` to `,`? I'm not sure how important it is to change internal name to verbose names, this is a backend interface. Maybe we can get feedback from the project team on that. 

If we do want verbose names, then should probably either iterate over set flags or maybe use simple str as the starting point.

Reply:
Also, if we do decide we want verbose names - can we calculate them once and store rather than converting every time `str` is called?

Reply:
Thanks for this re: `to_simple_str`.